ui/selectors/assets.ts

```diff
@@ -40,3 +59,152 @@ export function getAssetsMetadata(state: AssetsState) {
+          ? TEST_CHAINS.includes(chainId as (typeof TEST_CHAINS)[number])
+          : !TEST_CHAINS.includes(chainId as (typeof TEST_CHAINS)[number]),
+      ),
+    );
+    const tokensWithBalance: TokenWithFiatAmount[] = [];
+    Object.entries(filteredAccountTokensChains).forEach(
+      ([stringChainKey, tokens]) => {
+        const chainId = stringChainKey as Hex;
+        const tokenList = tokens as Token[];
+        tokenList.forEach((token: Token) => {
+          const { isNative, address, decimals } = token;
+          const balance =
+            calculateTokenBalance({
+              isNative,
+              chainId,
+              address,
+              decimals,
+              nativeBalances,
+              selectedAccountTokenBalancesAcrossChains,
+            }) || '0';
+
+          const tokenFiatAmount = calculateTokenFiatAmount({
+            token,
+            chainId,
+            balance,
+            marketData,
+            currencyRates,
+          });
+
+          // Respect the "hide zero balance" setting (when true):
+          // - Native tokens should always display with zero balance when on the current network filter.
+          // - Native tokens should not display with zero balance when on all networks filter
+          // - ERC20 tokens with zero balances should respect the setting on both the current and all networks.
+
+          // Respect the "hide zero balance" setting (when false):
+          // - Native tokens should always display with zero balance when on the current network filter.
+          // - Native tokens should always display with zero balance when on all networks filter
+          // - ERC20 tokens always display with zero balance on both the current and all networks filter.
+          if (
+            !hideZeroBalanceTokens ||
+            balance !== '0' ||
+            (token.isNative && isOnCurrentNetwork)
+          ) {
+            tokensWithBalance.push({
+              ...token,
+              balance,
+              tokenFiatAmount,
+              chainId,
+              string: String(balance),
+              primary: '',
+              secondary: 0,
+              title: '',
+            });
+          }
+        });
+      },
+    );
+    return tokensWithBalance;
+  },
+);
+
+export const getMultiChainAssets = createDeepEqualSelector(
+  (_state, selectedAccount) => selectedAccount,
+  getMultichainBalances,
+  getAccountAssets,
+  getAssetsMetadata,
+  getAssetsRates,
+  (_state) => _state.metamask.conversionRates,
+  (
+    selectedAccountAddress,
+    multichainBalances,
+    accountAssets,
+    assetsMetadata,
+    assetRates,
+    multichainCoinRates,
+  ) => {
+    const assetIds = accountAssets?.[selectedAccountAddress.id] || [];
+    const balances = multichainBalances?.[selectedAccountAddress.id];
+    return assetIds.map((assetId: CaipAssetId) => {
+      const [chainId, assetDetails] = assetId.split('/');
+      const isNative = assetDetails.split(':')[0] === 'slip44';
```

Comment:
Probably worth using the util for this: https://github.com/MetaMask/utils/blob/cba07ff6fd0fe57b3a41063fbf81f990a09b3590/src/caip-types.ts#L309

Reply:
thnx! created this https://github.com/MetaMask/metamask-extension/pull/30467 to bump the utils package

Reply:
updated here [4c6583f](https://github.com/MetaMask/metamask-extension/pull/30389/commits/4c6583f78cfe5fa571a14a1ffa971f6295d197e2)